File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/CalcFinalDamageValueMessage.ts
import {IMessage} from "Core/ObserverMessageKit";
import DamageInfo from "mods/ModTheSpire/Scripts/DataDefine/DamageInfo";
import AbstractCreature from "mods/ModTheSpire/Scripts/Unit/AbstractCreature";

//计算伤害结束
export class CalcFinalDamageValueMessage implements  IMessage{
    public static readonly Id = "CalcFinalDamageValue";
    public constructor(public info:DamageInfo,public source:AbstractCreature,public target: AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/CalcFinalDamageValueMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///计算最终伤害
    public class CalcFinalDamageValueMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Source;
        public AbstractCreature Target;

        public CalcFinalDamageValueMessage(DamageInfo damageInfo, AbstractCreature source, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Source = source;
            Target = target;
        }
    }
}